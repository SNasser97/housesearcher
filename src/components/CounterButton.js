import React, { Component } from "react";
//! //! performance demo 2/2
class CounterButten extends Component {
  constructor() {
    super();
    this.state = {
      count: 0
    }
  }
  shouldComponentUpdate(nextProps, nextState) {
    console.log(nextProps, nextState);
    return (this.state.count !== nextState.count);
  }
  updateCount = () => {
    this.setState(state => {
      return { count: state.count + 1 }
    });
  }
  render() {
    console.log('COUNTER BUTTON');
    return (
      <button style={{display:"block", margin:"0 auto"}}colour={ this.props.colour } onClick={ this.updateCount }> 
        COUNT: {this.state.count}
      </button>
    );
  }
}

export default CounterButten;