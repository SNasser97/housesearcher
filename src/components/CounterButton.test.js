import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CounterButton from './CounterButton';

// dummy_data from api (obj data that includes 4 objects)
const mockColour = 'red';
const wrapper = shallow(<CounterButton colour={mockColour} />);

//* snapshot test example
describe('Counter component', () => {
  it('expect to render counter', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('increment counter correctly', () => {
    // simulate a click of the counter
    wrapper.find('[test="ctr-button"]').simulate('click');
    wrapper.find('[test="ctr-button"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().colour).toEqual('red');
  });
});


// describe('<Card props={obj}...house_features={obj.obj.props} />', () => {
//   const wrapper = shallow(<Card features={features.house_features} />);

//   it('expect to render card', () => {
//     expect(wrapper.length)
//       .toEqual(1);
//   });

//   it('expect to contain description p element', () => {
//     const wrapper = shallow(<Card features = {features.house_features}>
//       <p className='house__description'></p>
//     </Card>);
//     expect(wrapper.contains(<p className='house__description'></p>
// )).toEqual(true);
//     console.log(wrapper.children());
//   });


// });

