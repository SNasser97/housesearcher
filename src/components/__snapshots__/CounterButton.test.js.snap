// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component expect to render counter 1`] = `
<button
  colour="red"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
      "margin": "0 auto",
    }
  }
  test="ctr-button"
>
  COUNT: 
  0
</button>
`;
